esphome/components/emporia_vue/sensor.py

```diff
@@ -0,0 +1,143 @@
+from esphome.const import (
+    CONF_CALIBRATION,
+    CONF_ID,
+    CONF_INPUT,
+    CONF_OTA,
+    CONF_VOLTAGE,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_VOLTAGE,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_WATT,
+    UNIT_VOLT,
+)
+
+CONF_CT_CLAMPS = "ct_clamps"
+CONF_PHASES = "phases"
+CONF_PHASE_ID = "phase_id"
+CONF_SENSOR_POLL_INTERVAL = "sensor_poll_interval"
+
+CODEOWNERS = ["@flaviut", "@Maelstrom96", "@krconv"]
+ESP_PLATFORMS = ["esp-idf"]
+DEPENDENCIES = ["i2c"]
+AUTOLOAD = ["sensor"]
+
+emporia_vue_ns = cg.esphome_ns.namespace("emporia_vue")
+EmporiaVueComponent = emporia_vue_ns.class_(
+    "EmporiaVueComponent", cg.Component, i2c.I2CDevice
+)
+PhaseConfig = emporia_vue_ns.class_("PhaseConfig")
+CTSensor = emporia_vue_ns.class_("CTSensor", sensor.Sensor)
+
+PhaseInputWire = emporia_vue_ns.enum("PhaseInputWire")
+PHASE_INPUT = {
+    "BLACK": PhaseInputWire.BLACK,
+    "RED": PhaseInputWire.RED,
+    "BLUE": PhaseInputWire.BLUE,
+}
+
+CTInputPort = emporia_vue_ns.enum("CTInputPort")
+CT_INPUT = {
+    "A": CTInputPort.A,
+    "B": CTInputPort.B,
+    "C": CTInputPort.C,
+    "1": CTInputPort.ONE,
+    "2": CTInputPort.TWO,
+    "3": CTInputPort.THREE,
+    "4": CTInputPort.FOUR,
+    "5": CTInputPort.FIVE,
+    "6": CTInputPort.SIX,
+    "7": CTInputPort.SEVEN,
+    "8": CTInputPort.EIGHT,
+    "9": CTInputPort.NINE,
+    "10": CTInputPort.TEN,
+    "11": CTInputPort.ELEVEN,
+    "12": CTInputPort.TWELVE,
+    "13": CTInputPort.THIRTEEN,
+    "14": CTInputPort.FOURTEEN,
+    "15": CTInputPort.FIFTEEN,
+    "16": CTInputPort.SIXTEEN,
+}
+
+SCHEMA_CT_CLAMP = sensor.sensor_schema(
+    unit_of_measurement=UNIT_WATT,
+    device_class=DEVICE_CLASS_ENERGY,
+    state_class=STATE_CLASS_MEASUREMENT,
+).extend(
+    {
+        cv.GenerateID(): cv.declare_id(CTSensor),
+        cv.Required(CONF_PHASE_ID): cv.use_id(PhaseConfig),
+        cv.Required(CONF_INPUT): cv.enum(CT_INPUT),
+    }
+)
+
+CONFIG_SCHEMA = cv.All(
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(EmporiaVueComponent),
+            cv.OnlyWith(CONF_OTA, "ota"): cv.use_id(OTAComponent),
+            cv.Optional(
+                CONF_SENSOR_POLL_INTERVAL, default="240ms"
+            ): cv.positive_time_period_milliseconds,
+            cv.Required(CONF_PHASES): cv.ensure_list(
+                {
+                    cv.Required(CONF_ID): cv.declare_id(PhaseConfig),
+                    cv.Required(CONF_INPUT): cv.enum(PHASE_INPUT),
+                    cv.Optional(CONF_CALIBRATION, default=0.022): cv.zero_to_one_float,
+                    cv.Optional(CONF_VOLTAGE): sensor.sensor_schema(
+                        unit_of_measurement=UNIT_VOLT,
+                        device_class=DEVICE_CLASS_VOLTAGE,
+                        state_class=STATE_CLASS_MEASUREMENT,
+                        accuracy_decimals=1,
+                    ),
+                }
+            ),
+            cv.Required(CONF_CT_CLAMPS): cv.ensure_list(SCHEMA_CT_CLAMP),
```

Comment:
I think nesting this sensor as a sub-object under the clamps config will help with future compatibility. We don't currently give a way to use the current sensor data, which I think we will eventually want to support. Nesting this sensor would help ensure we don't need to make a breaking change when we do add that data.